### PR TITLE
Add documentation for DNS stub domains

### PR DESCRIPTION
**Pull request by bowei**
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> For 1.7 Features: set Milestone to `1.7` and Base Branch to `release-1.7`
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>
> NOTE: Please check the “Allow edits from maintainers” box below to allow 
> reviewers fix problems on your patch and speed up the review process.
> Please delete this note before submitting the pull request.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/4063)

### CODE REVIEW

**Comment by bowei** [comment]
@dchen1107 needs 1.7 milestone

**Comment by chenopis** [comment]
@bowei 

1. You will need to use the Task template for this doc. You can use [`/docs/tasks/example-task-template.md`](https://raw.githubusercontent.com/kubernetes/kubernetes.github.io/master/docs/tasks/example-task-template.md) as an example.

2. Also, you will need to add an entry for this doc in the Tasks table of contents, [`/_data/tasks.yml`](https://github.com/kubernetes/kubernetes.github.io/blob/master/_data/tasks.yml)

**Comment by chenopis** [comment]
@bowei Who can we get to do the tech review?

FYI, all feedback must be addressed and LGTMs given by EOD Tue, June 27th so that this can be merged for the 1.7 release on June 28th.

**Comment by bowei** [comment]
updated

**Comment by chenopis** [comment]
@abiogenesis-now If you have time, here's one more. =)

File: docs/tasks/administer-cluster/dns-custom-nameservers.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+the DNS caching layer in kube-dns. From here, the suffix of the request is
+examined and then forwarded to the appropriate DNS.  In this case, names with
+the cluster suffix (e.g.; “.cluster.local”) are sent to kube-dns. Names with
+the stub domain suffix (e.g.; “.acme.local”) will be sent to the configured
+custom resolver. Finally, requests that do not match any of those suffixes will
+be forwarded to the upstream DNS.
+
+![DNS lookup flow](/docs/tasks/administer-cluster/dns-custom-nameservers/dns.png)
+
+## Configuring stub-domain and upstream DNS servers
```
Comment (MrHohn) [inline docs/tasks/administer-cluster/dns-custom-nameservers.md]: Do we need to mention which kube-dns version starts supporting this? May be confusing if user try it on old kube-dns.

File: docs/tasks/administer-cluster/dns-custom-nameservers.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+{% capture discussion %}
+
+## Understanding custom DNS upstream servers and stub domains
+
+### Pod DNS policies
+
+Kubernetes currently supports two DNS policies specified on a per-pod basis
+using the dnsPolicy flag: “Default” and “ClusterFirst”. If dnsPolicy is not
+explicitly specified, then “ClusterFirst” is used:
```
Comment (MrHohn) [inline docs/tasks/administer-cluster/dns-custom-nameservers.md]: An unrelated comment: Making "ClusterFirst" the default value while having another option called "Default" looks really confusion :)

File: docs/tasks/administer-cluster/dns-custom-nameservers.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+If dnsPolicy is set to “Default”, then the name resolution configuration is
+inherited from the node the pods run on. Note: this feature cannot be used in
```
Comment (MrHohn) [inline docs/tasks/administer-cluster/dns-custom-nameservers.md]: It seems to me "this feature" is a bit obscure. May be just call out the feature name?

File: docs/tasks/administer-cluster/dns-custom-nameservers.md (hunk 1 above)
Comment (bowei) [inline docs/tasks/administer-cluster/dns-custom-nameservers.md]: done

File: docs/tasks/administer-cluster/dns-custom-nameservers.md (hunk 2 above)
Comment (bowei) [inline docs/tasks/administer-cluster/dns-custom-nameservers.md]: done

File: docs/tasks/administer-cluster/dns-custom-nameservers.md (hunk 3 above)
Comment (bowei) [inline docs/tasks/administer-cluster/dns-custom-nameservers.md]: done

**Comment by bowei** [comment]
@MrHohn : addressed your comments

**Comment by MrHohn** [comment]
Thanks!

/lgtm

**Comment by abiogenesis-now** [comment]
won't be able to get to this until later tonight, but generally looks reasonable

**Comment by chenopis** [comment]
@abiogenesis-now No worries. I got it. Thx!

**Comment by abiogenesis-now** [comment]
@chenopis @bowei I know that this already merged and that the version cut is tonight; just wanted to make sure that it was intentional to have duplication in two sections?

Both *Name resolution in Kubernetes* and *Configuring stub-domain and upstream DNS servers* have the exact same paragraph that starts:
> The diagram below shows the flow of DNS queries specified in the configuration above...

However, the second section is missing the image `docs/tasks/administer-cluster/dns-custom-nameservers/dns.png`.

Is this intentional?

**Comment by chenopis** [comment]
@abiogenesis-now Actually, the version for the release won't be cut until tomorrow at the earliest. We can still put in a patch PR to make any changes for this. Great catch, thanks for taking a look!

Originally, there was a second image, so I'm guessing that's where it went. However, since it's not there right now, I'll take out the superfluous "The diagram below..." sentence.

@bowei Feel free to put in another PR if you want to add back the second image.

**Pull request merged**